### PR TITLE
feat(reborn): wire product auth composition seam

### LINKED ISSUE #3811: [Reborn] Step 2: Wire Reborn-native product auth and secrets composition
Parent: #3289
Depends on: #3810

## Purpose

Implement the second slice of #3289 after the Step 1 contract/fake-service foundation: wire the product-auth contract into a Reborn-native auth/secrets composition seam.

This step should create the production Reborn service boundary that future WebUI/setup/extension surfaces can call. It must not bridge Reborn product auth into V1 binary routes, V1 pending maps, V1 extension manager authority, or V1 secret-store semantics.

Implementation PRs for this issue should target `reborn-integration`.

## Scope

- Introduce a single discoverable Reborn composition point for `AuthFlowManager`, `AuthInteractionService`, `CredentialSetupService`, `CredentialAccountService`, `SecretCleanupService`, `AuthProviderClient`, and OAuth egress.
- Compose these services with Reborn-owned secrets, network/egress, no-exposure, continuation, and readiness boundaries as those substrate contracts become available.
- Keep durable auth-flow records, credential accounts, and secure manual-token interactions as typed Reborn product-auth records, not V1 pending state.
- Add redacted projection helpers for auth-required/manual-token/OAuth/account-selection states.
- Add caller-level tests around at least one Reborn composition/product-auth entrypoint that exercises the typed boundary rather than only testing helper functions.
- Add architecture guardrails if needed to prevent product-auth composition from depending on V1 route handlers, V1 `ExtensionManager` authority, raw `SecretsStore::get_decrypted`, raw `reqwest`, or route-local lifecycle/turn replay logic.

## Out of scope

- Do not wire Reborn product auth through V1-compatible web/chat/setup flows.
- Do not rewrite V1 OAuth callback routes in this step.
- Do not replace low-level durable secret encryption/storage unless that substrate is explicitly ready.
- Do not change runtime HTTP credential injection or one-shot lease behavior; those remain owned by #3068/#3088/#3140-style substrate work.
- Do not remove V1 pending maps from current production behavior in this slice.

## Acceptance criteria

- [ ] Production Reborn code has a single discoverable product-auth service composition seam.
- [ ] Reborn auth-flow, credential-account, secure-submit, cleanup, provider-client, and continuation contracts are composed without relying on V1 as product authority.
- [ ] Manual-token auth-required projections are redacted and do not expose token values to transcript/model/tool args/logs.
- [ ] OAuth provider exchange and refresh use the narrow Reborn auth provider/egress boundary, not route-local raw HTTP clients.
- [ ] At least one caller-level test verifies a Reborn caller enters through the typed product-auth boundary.
- [ ] Docs note which legacy V1 paths remain unmigrated evidence only and which Reborn-native seams are ready for future route/callback integration.

## Notes

This issue replaces the earlier “V1-compatible seam” framing. Reborn has its own auth/secrets mechanics and should be wired through the Reborn binary/composition path, not through V1 components.

### PR DESCRIPTION
**Pull request by hanakannzashi**
## Summary

- Adds `RebornProductAuthServices` as the single Reborn composition seam for product auth flows, secure manual-token interactions, credential setup/accounts, provider exchange, and cleanup.
- Wires the seam into `build_reborn_services` readiness: local-dev gets the in-memory Reborn auth implementation, while production only reports product-auth ready when a Reborn-native bundle is explicitly injected.
- Adds caller-level composition coverage for manual-token submit redaction and updates architecture guardrails/docs to keep #3811 separate from #3094 blocked-run gate resolution.

## Change Type

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [x] Documentation
- [ ] CI/Infrastructure
- [x] Security
- [ ] Dependencies

## Linked Issue

Closes #3811
Related #3289
Stacked on #3865 (`feat/reborn-auth-product-contracts`). After #3865 lands, this PR should be retargeted to `reborn-integration`.

## Validation

- [x] `cargo fmt --all -- --check`
- [ ] `cargo clippy --all --benches --tests --examples --all-features -- -D warnings`
- [ ] `cargo build`
- [x] Relevant tests pass: `cargo test -p ironclaw_reborn_composition --locked`, `cargo test -p ironclaw_auth --locked`, `cargo test -p ironclaw_architecture reborn_product_auth_contract_stays_reborn_native --locked`, `cargo test -p ironclaw_architecture reborn_crate_dependency_boundaries_hold --locked`, `cargo test -p ironclaw_architecture no_substrate_crate_depends_on_composition_root --locked`
- [ ] `cargo test --features integration` if database-backed or integration behavior changed
- [x] Manual testing: Not applicable; composition and contract behavior are covered by tests.
- [ ] If a coding agent was used and supports it, `review-pr` or `pr-shepherd --fix` was run before requesting review

Additional validation:
- `cargo clippy -p ironclaw_reborn_composition --all-targets -- -D warnings`
- `cargo check -p ironclaw_reborn_composition --features libsql --locked`
- `cargo check -p ironclaw_reborn_composition --features postgres --locked`
- `git diff --check`

## Security Impact

Yes. This adds the Reborn-native product-auth composition seam. It keeps callers on trait-shaped product-auth ports and documents that production must inject durable Reborn auth services explicitly instead of falling back to V1 routes, V1 pending maps, V1 `ExtensionManager`, V1 secret stores, or route-local raw HTTP clients. No production OAuth route, listener, raw provider transport, durable secret storage, token material handling, runtime credential injection, or database schema is added in this slice.

## Database Impact

None. This wires composition/readiness and tests only; durable product-auth storage remains future substrate work.

## Blast Radius

Limited to `ironclaw_reborn_composition`, the product-auth contract docs, and architecture guardrails. Existing V1 auth behavior is untouched. Local-dev now reports product-auth readiness through the in-memory Reborn auth implementation; production reports it only when explicitly injected.

## Rollback Plan

Revert this PR to remove the composition bundle, readiness flag, docs, and guardrail updates. No data migration or runtime cleanup is required.

## Review Follow-Through

Reviewer judgment requested on the exact composition surface and readiness semantics. #3094 remains separate: blocked run-state approval/auth gate listing, user decisions, and trusted resume should consume this auth boundary rather than adding a second auth model.

---

**Review track**: C

### CODE REVIEW

**Review by henrypark133 — COMMENTED** [review]
## Code Review (multi-agent)

**Intent:** Wire RebornProductAuthServices composition seam into service readiness with coverage, docs, and architecture guardrail updates.

**Stats:** 5 findings (from 5 raw, 5 after dedup) across 4 files. Reviewers run: security, bugs, performance, tests, conventions, design. Reviewers failed: none. Body-only: 0

### Tests

1. **Medium** — RebornProductAuthServices::new public constructor has no direct test (`crates/ironclaw_reborn_composition/src/auth.rs:46-62`, confidence 75) — anchor: crates/ironclaw_reborn_composition/src/auth.rs:46
   The public constructor taking six separate trait-object Arcs is never called directly in any test. The doc comment states production should prefer this over from_shared, but no test exercises it with distinct mock implementations.

2. **Medium** — RebornProductAuthServices::from_shared has no test with a custom multi-trait impl (`crates/ironclaw_reborn_composition/src/auth.rs:70-95`, confidence 75) — anchor: crates/ironclaw_reborn_composition/src/auth.rs:70
   from_shared is only exercised indirectly via local_dev_in_memory() with InMemoryAuthProductServices. No test verifies that from_shared correctly clones the Arc for each trait slot when given a custom type implementing all six traits.

3. **Medium** — RebornBuildInput::with_product_auth_services builder method has no test (`crates/ironclaw_reborn_composition/src/input.rs:171-176`, confidence 75) — anchor: crates/ironclaw_reborn_composition/src/input.rs:171
   The public builder method for injecting custom product-auth services into production builds is never exercised. No test verifies that a production build with injected services reports product_auth readiness as true and returns the injected bundle.

4. **Low** — Production build readiness with product_auth=false is not asserted in existing tests (`crates/ironclaw_reborn_composition/src/factory.rs:519-527`, confidence 50) — anchor: crates/ironclaw_reborn_composition/src/factory.rs:522
   Existing production tests do not assert that readiness.facades.product_auth is false when no auth services are injected, nor that services.product_auth is None.

5. **Low** — collect_forbidden_reborn_auth_file_uses helper has no dedicated test (`crates/ironclaw_architecture/tests/reborn_dependency_boundaries.rs:2262-2283`, confidence 50) — anchor: crates/ironclaw_architecture/tests/reborn_dependency_boundaries.rs:2262
   The new helper uses .expect() on file read and iterates forbidden patterns. No test verifies it correctly detects a violation when auth.rs contains a forbidden pattern, or that it handles a missing file path gracefully.

File: crates/ironclaw_reborn_composition/src/auth.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                &"Arc<dyn CredentialAccountService>",
+            )
+            .field("provider_client", &"Arc<dyn AuthProviderClient>")
+            .field("cleanup_service", &"Arc<dyn SecretCleanupService>")
+            .finish()
+    }
+}
+
+impl RebornProductAuthServices {
+    pub fn new(
```
Comment (henrypark133) [inline crates/ironclaw_reborn_composition/src/auth.rs:46]: **Medium** — RebornProductAuthServices::new public constructor has no direct test.

The public constructor taking six separate trait-object Arcs is never called directly in any test. The doc comment states production should prefer this over from_shared, but no test exercises it with distinct mock implementations.

**Fix:** tests::auth::reborn_product_auth_services_new_accepts_separate_impls covering distinct-trait-objects

File: crates/ironclaw_reborn_composition/src/auth.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        }
+    }
+
+    /// Builds a bundle from one object that implements every product-auth port.
+    ///
+    /// This is primarily for unified fakes such as
+    /// [`InMemoryAuthProductServices`]. Production composition should prefer
+    /// [`Self::new`] so storage, provider egress, interaction, and cleanup can
+    /// be supplied by separate implementations.
+    pub fn from_shared<T>(services: Arc<T>) -> Self
```
Comment (henrypark133) [inline crates/ironclaw_reborn_composition/src/auth.rs:70]: **Medium** — RebornProductAuthServices::from_shared has no test with a custom multi-trait impl.

from_shared is only exercised indirectly via local_dev_in_memory() with InMemoryAuthProductServices. No test verifies that from_shared correctly clones the Arc for each trait slot when given a custom type implementing all six traits.

**Fix:** tests::auth::reborn_product_auth_services_from_shared_clones_arc_per_trait covering multi-trait-clone

File: crates/ironclaw_reborn_composition/src/input.rs [hunk 3]
```diff
@@ -162,6 +163,16 @@ impl RebornBuildInput {
         self
     }
 
+    /// Inject a Reborn-native product-auth composition bundle.
+    ///
+    /// Production callers should provide durable implementations here once the
+    /// auth-flow and credential-account storage substrate is available. The
+    /// composition root never falls back to V1 route state or V1 secret stores.
+    pub fn with_product_auth_services(mut self, services: Arc<RebornProductAuthServices>) -> Self {
```
Comment (henrypark133) [inline crates/ironclaw_reborn_composition/src/input.rs:171]: **Medium** — RebornBuildInput::with_product_auth_services builder method has no test.

The public builder method for injecting custom product-auth services into production builds is never exercised. No test verifies that a production build with injected services reports product_auth readiness as true and returns the injected bundle.

**Fix:** tests::facade_factory::production_with_injected_product_auth_reports_ready covering injected-auth-readiness

File: crates/ironclaw_reborn_composition/src/factory.rs [hunk 4]
```diff
@@ -352,16 +372,13 @@ async fn build_production_shaped(
             // TODO(process-port): if production enables FirstParty runtime,
             // HostRuntimeServices must be given a production process port;
             // otherwise the LocalHostProcessPort default is rejected.
-            build_libsql_production(
+            let context = RebornProductionBuildContext {
```
Comment (henrypark133) [inline crates/ironclaw_reborn_composition/src/factory.rs:375]: **Low** — Production build readiness with product_auth=false is not asserted in existing tests.

Existing production tests do not assert that readiness.facades.product_auth is false when no auth services are injected, nor that services.product_auth is None.

**Fix:** tests::facade_factory::production_without_injected_auth_reports_product_auth_false covering default-auth-absent

File: crates/ironclaw_architecture/tests/reborn_dependency_boundaries.rs [hunk 5]
```diff
@@ -2252,3 +2258,26 @@ fn collect_forbidden_uses(
         }
     }
 }
+
+fn collect_forbidden_reborn_auth_file_uses(
```
Comment (henrypark133) [inline crates/ironclaw_architecture/tests/reborn_dependency_boundaries.rs:2262]: **Low** — collect_forbidden_reborn_auth_file_uses helper has no dedicated test.

The new helper uses .expect() on file read and iterates forbidden patterns. No test verifies it correctly detects a violation when auth.rs contains a forbidden pattern, or that it handles a missing file path gracefully.

**Fix:** tests::reborn_dependency_boundaries::collect_forbidden_reborn_auth_file_uses_detects_violation covering violation-detection
